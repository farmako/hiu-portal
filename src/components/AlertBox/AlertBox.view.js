import React from 'react';
import * as PropTypes from 'prop-types';
import { Alert, AlertTitle } from '@material-ui/lab';

const AlertBox = ({ type, title, message }) => (
  <Alert severity={type}>
    <AlertTitle>
      <strong>{title}</strong>
    </AlertTitle>
    {message}
  </Alert>
);

AlertBox.propTypes = {
  type: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  message: PropTypes.string,
};

AlertBox.defaultProps = {
  message: '',
};

export default AlertBox;
