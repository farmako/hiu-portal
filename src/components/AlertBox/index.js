import AlertBox from './AlertBox.view';

export default AlertBox;
