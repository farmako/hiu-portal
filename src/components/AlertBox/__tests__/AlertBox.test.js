import React from 'react';
import { shallow } from 'enzyme';
import { Alert, AlertTitle } from '@material-ui/lab';
import AlertBox from '../AlertBox.view';

describe('AlertBox', () => {
  const wrapper = shallow(
    <AlertBox
      type="info"
      title="Information"
      message="Health information is available!"
    />,
  );


  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show title and message in alert box', () => {
    expect(wrapper.find(AlertTitle).text()).toContain('Information');
    expect(wrapper.find(Alert).text()).toContain('Health information is available!');
  });
});
