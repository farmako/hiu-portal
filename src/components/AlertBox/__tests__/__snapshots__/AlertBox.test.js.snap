// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBox should render properly 1`] = `
<WithStyles(ForwardRef(Alert))
  severity="info"
>
  <WithStyles(ForwardRef(AlertTitle))>
    <strong>
      Information
    </strong>
  </WithStyles(ForwardRef(AlertTitle))>
  Health information is available!
</WithStyles(ForwardRef(Alert))>
`;
